videostore/renamer.go

```diff
@@ -0,0 +1,140 @@
+package videostore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"go.viam.com/rdk/logging"
+)
+
+// renamer watches a directory and converts local timestamps to unix timestamps
+type renamer struct {
+	watchDir     string
+	outputDir    string
+	pendingFiles []string
+	processLock  sync.Mutex
+	logger       logging.Logger
+}
+
+func newRenamer(watchDir, outputDir string, logger logging.Logger) *renamer {
+	return &renamer{
+		watchDir:     watchDir,
+		outputDir:    outputDir,
+		pendingFiles: []string{},
+		logger:       logger,
+	}
+}
+
+// processSegments watches the directory and processes new files
+//
+// This function loops indefinitely, watching for new MP4 files created
+// in the specified directory. When a new file is detected, it is queued for processing.
+// We do not need to spawn routines to process events as the segmenter will only produce
+// events every 30 seconds.
+func (r *renamer) processSegments(ctx context.Context) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return fmt.Errorf("failed to create watcher: %w", err)
+	}
+	defer watcher.Close()
+	if err := watcher.Add(r.watchDir); err != nil {
+		return fmt.Errorf("failed to add directory to watcher: %w", err)
+	}
+	r.logger.Debugf("starting to watch directory:", r.watchDir)
+	for {
+		select {
+		case event, ok := <-watcher.Events:
+			if !ok {
+				return errors.New("watcher closed unexpectedly")
+			}
+			// Only process file creation events for MP4 files
+			if event.Op&fsnotify.Create == fsnotify.Create && strings.HasSuffix(event.Name, ".mp4") {
+				r.logger.Debug("mp4 file created:", event.Name)
+				r.queueFile(event.Name)
+			}
+		case err, ok := <-watcher.Errors:
+			if !ok {
+				return errors.New("watcher error channel closed unexpectedly")
+			}
+			return fmt.Errorf("watcher error: %w", err)
+		case <-ctx.Done():
+			return ctx.Err()
```

Comment:
it is not an error for the the context to be cancelled. This shouldn't be an error.

Reply:
Changed, we now actually flush files from this path now.